corelib-utils/src/main/java/eu/europeana/corelib/utils/DateUtils.java

```diff
@@ -53,10 +61,75 @@ public static String format(Date date) {
 	 */
 	public static Date parse(String date) {
 		try {
-			return formatter.parse(date);
+			return DATE_TIME_FORMATTER.parse(date);
 		} catch (ParseException e) {
-			LOG.error(e.getMessage());
+			LOG.warn(e.getMessage());
 		}
 		return null;
 	}
+
+	/**
+	 * Returns true if the string is a year  "yyyy"
+	 *
+	 * @param value value to check
+	 * @return true when the string is a year
+	 */
+	public static boolean isYear(String value) {
+		//check for single year ex: 1930
+		if (StringUtils.isNotEmpty(value)) {
+			try {
+				Year year = Year.of(Integer.parseInt(value));
+				return year != null;
+			} catch (Exception e) {
+				return false;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Returns true if the string is a Year Range "yyyy-yyy"
+	 *
+	 * @param value value to check
+	 * @return true when the string is a year range
+	 */
+	public static boolean isYearRange(String value) {
+		String[] years = value.split("-");
+		//check for range ex: 1980-1990
+		if (years.length == 2) {
+			try {
+				return Year.parse(years[0]).isBefore(Year.parse(years[1]));
+			}
+			catch(Exception e) {
+				return false;
+			}
+		}
+		return false;
+	}
+	/**
+	 * Returns true if the string is a ISO8601 date (format: "yyyy-MM-dd")
+	 *
+	 * @param value value to check
+	 * @return true when the string is a ISO8601 date
+	 */
+	public static boolean isIsoDate(String value) {
+		try {
+			TemporalAccessor tmp = ISO_DATE_FORMATTER.parse(value);
+			return tmp != null;
+		} catch (Exception e) {
+			return false;
+		}
+	}
+
+	/**
+	 * Returns true if the string is a ISO8601 date time (format:
+	 * "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+	 *
+	 * @param value value to check
+	 * @return true when the string is a ISO8601 date time
+	 */
+	public static boolean isIsoDateTime(String value) {
+		return DateUtils.parse(value) != null;
```

Comment:
Why are we catching in other methods, but not here?

Reply:
this calls another method in DateUtils, which catches the exception and logs it too